packages/asset-buyer/src/quote_consumers/dynamic_swap_quote_consumer.ts

```diff
@@ -0,0 +1,117 @@
+import { ContractWrappers } from '@0x/contract-wrappers';
+import { BigNumber, providerUtils } from '@0x/utils';
+import { SupportedProvider, ZeroExProvider } from '@0x/web3-wrapper';
+import * as _ from 'lodash';
+
+import { constants } from '../constants';
+import {
+    CalldataInfo,
+    DynamicSwapQuoteExecutionOpts,
+    DynamicSwapQuoteGetOutputOpts,
+    ExchangeMarketBuySmartContractParams,
+    ExchangeMarketSellSmartContractParams,
+    ForwarderMarketBuySmartContractParams,
+    ForwarderMarketSellSmartContractParams,
+    SmartContractParamsInfo,
+    SwapQuote,
+    SwapQuoteConsumer,
+    SwapQuoteConsumerOpts,
+} from '../types';
+import { assert } from '../utils/assert';
+import { assetDataUtils } from '../utils/asset_data_utils';
+import { swapQuoteConsumerUtils } from '../utils/swap_quote_consumer_utils';
+
+import { ExchangeSwapQuoteConsumer } from './exchange_swap_quote_consumer';
+import { ForwarderSwapQuoteConsumer } from './forwarder_swap_quote_consumer';
+
+type SmartContractParams =
+    | ExchangeMarketBuySmartContractParams
+    | ExchangeMarketSellSmartContractParams
+    | ForwarderMarketBuySmartContractParams
```

Comment:
Can we export these unions from their respective classes and then have the final union here.